### PR TITLE
cartage-remote 2.0

### PR DESCRIPTION
**Pull request by halostatue**
- Rewrote for compatibility with cartage 2.0
- Configuration now supports multiple servers.

### CODE REVIEW

File: test/test_cartage_remote.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    it 'converts the implicit default into explicit' do
+      remote_config.delete(:hosts)
+      remote_config[:server] = hash_host
+
+      assert_equal remote_config[:hosts], cartage.config(for_plugin: :remote).
+        dig(:hosts).to_hash
+    end
+
+    it 'converts the implicit default into explicit' do
```
Comment (ravster) [inline test/test_cartage_remote.rb]: Duplicate of above test?

File: lib/cartage/plugins/remote.rb [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+      @stage = MicroMachine.new('new').tap { |stage|
+        stage.when :local_setup, 'new' => 'local_config'
+        stage.when :ssh_setup, 'local_config' => 'ssh_config'
+        stage.when :run_prebuild, 'ssh_config' => 'prebuild'
+        stage.when :clone_remote, 'prebuild' => 'remote_clone'
+        stage.when :build_remote, 'remote_clone' => 'remote_build'
+        stage.when :clean_remote, 'remote_build' => 'cleanup'
+        stage.when :complete, 'cleanup' => 'finished'
+        stage.on(:any) { |event| dispatch(event, stage.state) }
+      }
```
Comment (llwire) [inline lib/cartage/plugins/remote.rb:239]: This is coooool!
bitmoji

File: lib/cartage/plugins/remote.rb [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+      @config_file = make_config(paths).path
+
+      @subs = OpenStruct.new(
+        paths.to_h.merge(
+          repo_url:        cartage.repo_url,
+          name:            cartage.name,
+          release_hashref: cartage.release_hashref,
+          timestamp:       cartage.timestamp,
+          remote_host:     host.address,
+          remote_address:  host.address,
```
Comment (llwire) [inline lib/cartage/plugins/remote.rb:313]: `remote_host` and `remote_address` are the same.

Comment (halostatue) [inline lib/cartage/plugins/remote.rb:313]: That‘s actually intentional for backwards compatibility with older scripts. The proper name is `remote_address`, but older scripts may have `remote_host`.

**Comment by llwire** [comment]
+1

**Pull request merged**